api/jobs/handlers.py

```diff
@@ -129,6 +151,28 @@ def add(self):
             destination.check_access(self.uid, 'rw')
             now_flag = False # Only superuser requests are allowed to set "now" flag
 
+        # Config manifest check
+        gear = get_gear_by_name(gear_name)
+        if len(gear.get('manifest', {}).get('config', {})) > 0:
+
+            invocation = gear_tools.derive_invocation_schema(gear['manifest'])
+            ci = gear_tools.isolate_config_invocation(invocation)
+            validator = Draft4Validator(ci)
+
+            try:
+                validator.validate(config_)
+            except ValidationError as err:
+                key = None
+                if len(err.relative_path) > 0:
+                    key = err.relative_path[0]
+
+                self.response.set_status(422)
+                return {
+                    'reason': 'config did not match manifest',
+                    'error': err.message.replace("u'", "'"),
+                    'key': key
+                }
```

Comment:
This bit is a workaround for not being able to send back structured data in an error return. In the future, this could be a library call. I can document this if desired.